dash-spv-coinjoin/src/wallet_ex.rs

```diff
@@ -175,13 +190,74 @@ impl WalletEx {
         rounds_ref
     }
 
+    pub fn has_collateral_inputs(&mut self, only_confirmed: bool) -> bool {
+        unsafe { (self.has_collateral_inputs)(only_confirmed, self, self.opaque_context) }
+    }
+
+    pub fn select_coins_grouped_by_addresses(
+        &mut self, 
+        skip_denominated: bool, 
+        anonymizable: bool, 
+        skip_unconfirmed: bool, 
+        max_outpoints_per_address: i32
+    ) -> Vec<CompactTallyItem> {
+        // Try using the cache for already confirmed mixable inputs.
+        // This should only be used if maxOupointsPerAddress was NOT specified.
+        if max_outpoints_per_address == -1 && anonymizable && skip_unconfirmed {
+            if skip_denominated && self.anonymizable_tally_cached_non_denom {
+                println!("[RUST] CoinJoin: SelectCoinsGroupedByAddresses - using cache for non-denom inputs {}", self.vec_anonymizable_tally_cached_non_denom.len());
+                return self.vec_anonymizable_tally_cached_non_denom.clone();
```

Comment:
Not sure if there is a better way to cache Vec<CompactTallyItem> without cloning it.

Reply:
I think, it's ok here since the caller then will sort this